src/web/features.rs

```diff
@@ -0,0 +1,63 @@
+use crate::db::types::Feature;
+use crate::{
+    db::Pool,
+    impl_webpage,
+    web::{page::WebPage, MetaData},
+};
+use iron::{IronResult, Request, Response};
+use router::Router;
+use serde::Serialize;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
+struct FeaturesPage {
+    metadata: MetaData,
+    features: Option<Vec<Feature>>,
+    feature_len: usize,
+    default_feature_len: usize,
+}
+
+impl_webpage! {
+    FeaturesPage = "crate/features.html",
+}
+
+pub fn build_features_handler(req: &mut Request) -> IronResult<Response> {
+    let router = extension!(req, Router);
+    let name = cexpect!(req, router.find("name"));
+    let version = cexpect!(req, router.find("version"));
+
+    let mut conn = extension!(req, Pool).get()?;
+
+    let query = ctry!(
+        req,
+        conn.query(
+            "SELECT crates.name,
+                releases.features
+             FROM builds
+             INNER JOIN releases ON releases.id = builds.rid
+             INNER JOIN crates ON releases.crate_id = crates.id
+             WHERE crates.name = $1 AND releases.version = $2
+             ORDER BY releases.id DESC",
+            &[&name, &version]
+        )
+    );
+    let row = cexpect!(req, query.get(0));
+    let features = MetaData::parse_features(row.get("features"));
```

Comment:
This is already a field on `MetaData`, no need to parse and `SELECT` it twice. You can change everything using this to use `metadata.features` instead, then remove it from the SQL query.

Alternatively, since the /features page is the only one using this data, you can remove it from `MetaData` altogether and only have it for `/features`. But it definitely shouldn't be in both.

Reply:
Good idea, thanks